artifact/v3/payload/CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 # Test the parser
 add_executable(artifact_payload_parser_test EXCLUDE_FROM_ALL
   payload.cpp
+  ${CMAKE_SOURCE_DIR}/artifact/v3/manifest/manifest.cpp
+  ${CMAKE_SOURCE_DIR}/artifact/error.cpp
```

Comment:
Why not link to the lib instead?

Reply:
It has not lib. This is the `artifact` error stuff. Or do you mean link to the whole `artifact` target?

Reply:
Yes, the whole thing. It will only grab the symbols it needs.

Reply:
Yeh, I see what you mean. I will favour against it this time though, as then it will build the whole shebang, when only the error and manifest symbols are required.

Reply:
When building only that test, yes. But be aware that this increases the total build time, since you are building those files twice now.